sudo/src/run_handler.rs

```diff
@@ -191,6 +191,53 @@ pub fn run_target(
     do_request(req, copy_env, manually_requested_dir)
 }
 
+/// Returns all the extensions from PATHEXT that are used for "executables"
+fn get_path_extensions() -> Vec<String> {
+    env::var("PATHEXT")
+        .unwrap_or_else(|_| {
+            // If PATHEXT isn't set, use the default
+            ".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC".to_string()
+        })
+        .split(';')
+        .map(|ext| ext.to_string())
+        .collect()
+}
+
+/// Searches the PATH for the given filename, with the extensions from PATHEXT.
+/// Returns the full path to the file if it's found, or an error if it's not.
+fn search_path_with_extensions(filename: &str) -> Result<String> {
+    let filename = HSTRING::from(filename);
+    let mut buffer = vec![0; 260];
+    let extensions = get_path_extensions();
+    for extension in extensions {
+        let extension_hstring = HSTRING::from(extension);
+        // SearchPathW will ignore the extension if the filename already has one.
+        let len: u32 =
+            unsafe { SearchPathW(None, &filename, &extension_hstring, Some(&mut buffer), None) };
```

Comment:
I am fairly certain that SearchPathW doesn't take an `HSTRING` o_O

Is this a weird `rust-win32`ism?

Reply:
`SearchPathW` expects a 16-bit wide string but you're starting with a UTF-8 string. `HSTRING` provides a 16-bite wide string and comprehensive conversion from various Rust Standard Library types, so it's very convenient in cases like this.

Reply:
Unlike `BSTR`, `HSTRING` is not compatible when passed as a `LPWSTR`. It does not point directly to the character data. Why is this conversion OK, in contravention of all Windows API design to-date?

Reply:
(I am happy to accept “there are magic conversions in the rust Windows crate,” but that is exactly what I mean when I ask if it is a _weirdness_. I also want somebody to say it on-record.)

Reply:
I assume by `LPWSTR` you really mean `PCWSTR`. 😉 

In which case, `HSTRING` is actually a far better choice than `BSTR`. While `BSTR` is literally the pointer, that pointer is far more dangerous in general and `BSTR` semantics are far more nebulous than `HSTRING`. An `HSTRING` is perfectly compatible with `PCWSTR` and even has an API for that - `WindowsGetStringRawBuffer` - but I implemented it inline for both C++/WinRT and Rust and there is thus no performance penalty for pointer access. `HSTRING` is generally more efficient, can avoid heap allocations, and while [windows-strings](https://crates.io/crates/windows-strings) does provide all kinds of conveniences for both `HSTRING` and `BSTR`, the former is preferred.